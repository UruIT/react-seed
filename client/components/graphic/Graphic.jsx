import React from 'react';
import style from './Graphic.scss';
import Chart from '../chart/Chart';
import data from './graphic-data';

export default class Graphic extends React.Component {
	constructor(props) {
		super(props);

		this.handleClick = this.handleClick.bind(this);

		this.state = {
			selectedYear: 2019,
			data
		};
	}

	render() {
		const { data, selectedYear } = this.state;

		return (
			<div className={style.graphic}>
				<div className={style.chartContainer}>
					<Chart data={data} onClick={this.handleClick} />
				</div>
				<div className={style.infoContainer}>
					<div className={style.title}>
						{selectedYear}
					</div>
					{'This is the information related to year ' + selectedYear}
				</div>
			</div>
		);
	}

	componentDidMount() {
		fetch('/api/icon')
			.then(res => res.text())
			.then(text => {
				const image = new Image(40, 40);
				image.src = text;
				const data = this.state.data;
				let dataset = data.datasets[0];
				Object.assign(dataset, {
					pointStyle: ['', '', image, '', '', image, '', '', '', image, '', '', image]
				});
				this.setState({ data });
			})
			.catch(console.error);
	}

	handleClick(event) {
		const index = event.elements[0]._index;
		const selectedYear = this.state.data.labels[index];
		this.setState({ selectedYear });
	}
}
