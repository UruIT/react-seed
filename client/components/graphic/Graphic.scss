.graphic {
	text-align: center;
	display: flex;

	.chart-container {
		width: 70%;
		display: inline-block;
	}

	.info-container {
		border: solid 1px;
		display: inline-block;
		width: 25%;
		margin: auto;
		padding: 25px;
		text-align: left;

		.title {
			font-size: 40px;
			text-align: center;
			width: 100%;
			text-decoration: underline;
			margin-bottom: 25px;
		}
	}
}
