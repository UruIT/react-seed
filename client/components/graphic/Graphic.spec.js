import React from 'react';
import Graphic from './Graphic';
import ShallowRenderer from 'react-test-renderer/shallow';

test('renders Graphic', () => {
	const renderer = new ShallowRenderer();
	const tree = renderer.render(<Graphic />);

	expect(tree).toMatchSnapshot();
});
