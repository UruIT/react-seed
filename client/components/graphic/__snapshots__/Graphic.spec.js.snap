// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Graphic 1`] = `
<div
  className="graphic"
>
  <div
    className="chartContainer"
  >
    <Chart
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "#49b9ab",
              "borderColor": "#49b9ab",
              "data": Array [
                0,
                65500,
                95000,
                120000,
                130000,
                175000,
                200000,
                230000,
                260000,
                270000,
                300000,
                320000,
                350000,
              ],
              "fill": false,
              "hoverRadius": 6,
              "label": "Starting now",
              "radius": 3,
            },
            Object {
              "backgroundColor": "#FA5858",
              "borderColor": "#FA5858",
              "borderDash": Array [
                5,
                5,
              ],
              "data": Array [
                null,
                null,
                null,
                null,
                0,
                50000,
                60500,
                70000,
                75000,
                90000,
                120000,
                145000,
                150000,
              ],
              "fill": false,
              "hoverRadius": 6,
              "label": "Starting in five years",
              "radius": 3,
            },
          ],
          "labels": Array [
            2017,
            2018,
            2019,
            2020,
            2021,
            2022,
            2023,
            2024,
            2025,
            2026,
            2027,
            2028,
            2029,
          ],
        }
      }
      onClick={[Function]}
    />
  </div>
  <div
    className="infoContainer"
  >
    <div
      className="title"
    >
      2019
    </div>
    This is the information related to year 2019
  </div>
</div>
`;
